interview-questions.md

```diff
@@ -14,44 +14,44 @@ Researched answer:
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: A Boolean value is true or false, and the operators == (loosely equal), === (strictly equal), > (greater than), >= (greater than or qual to), < (less than), <= (less than or qual to), !== (strictly not equal) will return a Boolean value.
 
-Researched answer:
+Researched answer: There is also the != (loosely not equal) operator. We can combine conditions using these operators with && (AND) and || (OR). With &&, both conditions must be true for it to be true. With ||, one condition must be true for it to be true.
 
 3. What is an index? What is the difference between index and value?
 
-Your answer:
+Your answer: An index indicates the location of a value within an element. We indicate an index using brackets. Unlike counting with length, counting with indexes starts at 0. The index refers to the location while the value is the actual data that the index is referring to.
 
-Researched answer:
+Researched answer: The index tells us the position of a certain element. When we use index for strings, each number position will refer to a character within the string. For arrays, indexes will refer to the different values in it.
 
 4. What is iteration?
 
-Your answer:
+Your answer: Iteration is running through a code. In the case of for loops and while loops, programs will iterate through these loops until there is an indiction to stop.
 
-Researched answer:
+Researched answer: Iteration refers to the repeated execution of a code, and a code will be iterated through until the specified end result is reached. When a code block is repeated multiple times, that is called a loop. When the code is run through one time, that is a single iteration.
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-Your answer:
+Your answer: Pair programming consists of a driver and a navigator. The driver is the person typing out the code, and the navigator is the person telling the driver what to type out. The navigator should properly explain their thought processes and let the driver know why they are having the driver type out that code. Meanwhile, the driver should follow what the navigator says, but they should speak up if they have their own opinion about the code or maybe have their own approach.
 
-Researched answer:
+Researched answer: By having two people working on code, it is more likely to result in fewer defects. With the separate roles of the programmers, one person may have more experience or knowledge about a certain aspect of code and can navigate that part of the code, or if the navigator is too focused on one way of thinking, the driver may be able to help with a different perspective.
 
 ## Looking Ahead: Terms for Next Week
 
```

Comment:
It's always great to look ahead at the terms so you have some context when hearing the terms during lectures.